monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js

```diff
@@ -234,7 +236,11 @@ class ConfigurePageComponent extends AuthComponent {
 
   updateConfigSection = () => {
     let newConfig = this.state.configuration;
-    if (Object.keys(this.state.currentFormData).length > 0) {
+
+    if (this.currentSection === 'attack' && this.state.currentFormData === undefined) {
```

Comment:
Why do we need to check if `this.state.currentFormData === undefined`? how did it get `undefined`?

Reply:
![Undefined](https://user-images.githubusercontent.com/15820737/125949534-0c49a217-0b26-4d4f-9cce-70fd83f6bad1.gif)

Line 348. Because the key is `attack`. It is setting the `currentFormData` to `this.state.configuration[key]` which is `undefined`

Reply:
So maybe when setting the `attack` tab we should set form data to `{}` so the check of `Object.keys(this.state.currentFormData).length > 0` would work again?